Classes/Repository/RepositoryViewController.swift

```diff
@@ -81,6 +85,50 @@ NewIssueTableViewControllerDelegate {
         return URL(string: "https://github.com/\(repo.owner)/\(repo.name)")!
     }
 
+    var bookmark: Bookmark {
+        return Bookmark(
+            type: .repo,
+            name: repo.name,
+            owner: repo.owner,
+            hasIssueEnabled: repo.hasIssuesEnabled,
+            defaultBranch: repo.defaultBranch
+        )
+    }
+
+    func configureNavigationItems() {
+        guard let store = client.bookmarksStore else {
+            navigationItem.rightBarButtonItem = moreOptionsItem
+            return
+        }
+
+        let isNewBookmark = !store.contains(bookmark)
+        let bookmarkItem = UIBarButtonItem(
+            image: isNewBookmark ? UIImage(named: "nav-bookmark") : UIImage(named: "nav-bookmark-selected"),
+            style: .plain,
+            target: self,
+            action: #selector(IssuesViewController.toggleBookmark(sender:))
+        )
+        bookmarkItem.accessibilityLabel = isNewBookmark ? Constants.Strings.bookmark : Constants.Strings.removeBookmark
+        navigationItem.rightBarButtonItems = [moreOptionsItem, bookmarkItem]
+    }
+
+    @objc
+    func toggleBookmark(sender: UIBarButtonItem) {
+        guard let store = client.bookmarksStore else {
+            ToastManager.showGenericError()
+            return
+        }
+
+        if !store.contains(bookmark) {
+            store.add(bookmark)
+            sender.image = UIImage(named: "nav-bookmark-selected")
+        } else {
+            store.remove(bookmark)
+            sender.image = UIImage(named: "nav-bookmark")
+        }
+        Haptic.triggerNotification(.success)
```

Comment:
A lot of this toggling code looks identical to the stuff in Issues. Think you could make a new file with a shared function both call sites can use? Similar to what I did when this was an alert.

Reply:
Fair point 👍  Let me try!